hashintel/hash
# Pass plain text to text blocks

## 🌟 What is the purpose of this PR?




This is a blocker for our implementation of the [Hook Service](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0430-hook-service.md)

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1201095311341924/1202694273052402/f) _(internal)_

## 🚫 Blocked by




N/A

## 🔍 What does this change?




- Overwrite the text property on text blocks to be a calculated version of the plain text, instead of the rich text entity itself

## 📜 Does this require a change to the docs?



No

## ⚠️ Known issues



I don't calculate the text value of mentions at the moment. We need to do that later, but it's not really a blocker for the hook service and it's not trivial to do it 

## 🐾 Next steps



## 🛡 What tests cover this?



None 

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Make a few different text blocks
1.  Insert the graph properties being passed to see that they're the plain text
